oscarlab/graphene
# [Pal] Make "syscall" instruction appear only once in PAL binary

## Description of the changes 
This is needed to properly handle signals interrupting ocalls on SGX PAL. It will probably also be useful in fixing the issue of signals appearing while in LibOS, but before issuing a host level syscall instruction (in which case the signal is blocked until the syscall returns or another signal appears).

Fixes #2532

## How to test this PR? 
Check #2532 for example code failing on current master.


---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/gramineproject/graphene/2602)
